### PR TITLE
Add support for dark theme for mermaid diagrams

### LINKED ISSUE #4195: Mermaid.js diagrams don't work with the dark theme
**To Reproduce**
1. Create mermaid diagram
2. Select dark theme for outline

**Expected behavior**
Mermaid diagram changes the text color to white

**Screenshots**
image
image

**Outline (please complete the following information):**
- Install: self hosted
- Version: 0.66.1

**Desktop (please complete the following information):**
 - OS: linux
 - Browser Firefox
 - Version 103

### PR DESCRIPTION
**Pull request by not7cd**
This will fix #4195,
I'm able to pull the currently chosen theme. But I don't know how to trigger re-render when the theme is changed through the user menu.

### CODE REVIEW

File: shared/editor/plugins/Mermaid.ts [hunk 1]
```diff
@@ -66,6 +67,7 @@ function getNewState({
               htmlLabels: false,
             },
             fontFamily: "inherit",
+            theme: stores.ui.resolvedTheme === "light" ? "default" : "dark",
```
Comment (tommoor) [inline shared/editor/plugins/Mermaid.ts:70]: Ah, thanks for trying. Unfortunately you cannot import things from `~/` into `shared` as that's importing client dependencies into the server. I think you can pass it to the plugin from `node.editor.props.theme`, you can listen for changes with `window.addEventListener("theme-changed"`

Comment (not7cd) [reply shared/editor/plugins/Mermaid.ts:70]: I see that the same approach is used within `editor.components.ComponentView`. Also within this ComponentView I should have access to `node.editor.props.theme`?

**Comment by tommoor** [comment]
Thanks for the push – took this over the finish line here https://github.com/outline/outline/pull/4564

**Comment by not7cd** [comment]
Thanks! This definetly needed more understanding of the internals.

**Pull request closed without merging**